submit.py

```diff
@@ -67,50 +102,71 @@
-            predicted_probabilities = np.concatenate(probabilities_list).mean(axis=0)
-
-        else:
-            logits = model(X.unsqueeze(0).to(DEVICE))
-            predicted_probabilities = softmax(logits).cpu().detach().numpy()[0]
-        predicted_class = np.argmax(predicted_probabilities)
-
-        output_list.append(
-            {
-                "fid": fid,
-                "crop_id": predicted_class + 1,  # save label list
-                "crop_name": config["classes"][predicted_class],
-                "crop_probs": predicted_probabilities,
-            }
-        )
-
-output_frame = pd.DataFrame.from_dict(output_list)
-
-submission_path = Path(f"submissions/{name}/{pos}-submission-{name}.json")
-submission_path.parent.mkdir(parents=True, exist_ok=True)
-output_frame.to_json(submission_path)
-
-print(f"Saving: {name}.tar.gz ")
-
-os.system(f"cd submissions && tar czf {name}.tar.gz {name}")
+predictions = []
+groundtruths = []
+
+if model_type == "original":
+    with torch.no_grad():
+        for X, _, mask, fid in tqdm(
+            reader,
+            total=len(reader),
+            position=0,
+            leave=True,
+            desc="INFO: Saving predictions:",
+        ):
+            if config["spatial_backbone"] == "pixelsetencoder":
+                logits = model((X.unsqueeze(0).to(DEVICE), mask.unsqueeze(0).to(DEVICE)))
+                predicted_probabilities = softmax(logits).cpu().detach().numpy()[0]
+            elif config["spatial_backbone"] == "random_pixel":
+                # Ensemble the pixel predictions
+                model_input = torch.permute(X, (2, 0, 1)).to(DEVICE)
+                # Model input can get too large for GPU memory, so we need to split it up
+                # into chunks of size batch_size
+                model_input_chunks = torch.split(model_input, config["batch_size"], dim=0)
+                probabilities_list = []
+                for model_input_chunk in model_input_chunks:
+                    logits = model(model_input_chunk)
+                    predicted_probabilities = softmax(logits).cpu().detach().numpy()
+                    probabilities_list.append(predicted_probabilities)
+                predicted_probabilities = np.concatenate(probabilities_list).mean(axis=0)
+            else:
+                logits = model(X.unsqueeze(0).to(DEVICE))
+                predicted_probabilities = softmax(logits).cpu().detach().numpy()[0]
+            predicted_class = np.argmax(predicted_probabilities)
+
+            groundtruths.append(fid)
+            predictions.append(config["classes"][predicted_class])
+
+            output_list.append(
+                {
+                    "fid": fid,
+                    "crop_id": predicted_class + 1,  # save label list
+                    "crop_name": config["classes"][predicted_class],
+                    "crop_probs": predicted_probabilities,
+                }
+            )
+elif model_type == "temporal_augmentor":
+    pass
```

Comment:
unnecessary ^ elif

Reply:
This elif is to be updated with temporal_augmentor case.   It's just a placeholder in case I forget later.